File: 9-palindrome-number/9-palindrome-number.scala
object Solution {
    def isPalindrome(x: Int): Boolean = {
        val str = x.toString
        var res = true
        for(i <- 0 to (str.length / 2) - 1) {
            res &&= str.charAt(i) == str.charAt(str.length - 1 - i)
        }
        res
    }
}

File: 9-palindrome-number/9-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) return false;
        
        int n = 0;
        int k = x;
        
        while(k > 0) {
            n = n * 10 + k % 10;
            k /= 10;
        }
        
        return x == n;
    }
}